src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs

```diff
@@ -746,8 +746,7 @@ public Task ChangeDocumentAsync(DocumentId documentId, SourceText text)
 
         public Task ChangeDocumentAsync(DocumentId documentId, Solution solution)
         {
-            var oldSolution = this.CurrentSolution;
-            var newSolution = this.SetCurrentSolution(solution);
+            var (oldSolution, newSolution) = this.SetCurrentSolution(solution);
```

Comment:
new pattern is to use the data from SetCurrentSolution and use that to raise events.  It tells you the actual old/new solutions so it doesn't have to be inferred.